### ISSUE TITLE
Enable reviews is turned off in translations

### ISSUE THREAD

**Issue by Jon007** [hyyan/woo-poly-integration#216]
#### Can you reproduce this issue on default Wordpress theme (eg Storefront)?
Yes

#### Can you reproduce this issue when all other plugins are disabled except WooCommerce, Polylang and Hyyan WooCommerce Polylang Integration?
Yes

#### What product versions and settings are you using when this issue occurs?
* PHP: 7.0
* WordPress: 4.8 
* WooCommerce: 3.1
* Polylang: 2.1.5
* Hyyan WooCommerce Polylang Integration: 1.0.2
* Browser: any

#### Steps to Reproduce

1. check settings are set to synchronize comment status
Languages, Settings, Comment status
Settings, WooPoly, Metas List, Comment Status  _its unclear why this setting exists, since Comment Status is not a Meta, this should not have any effect_

2. pick an existing product with translation, check Product data, Advanced, “Enable reviews” is checked

3. Save the product (in any language), with Enable Reviews still selected

#### What I Expected
Enable Reviews should remain enabled in all translations

#### What Happened Instead
Enable Reviews stays enabled only in the product version which is being saved:  in all translations it is turned off

This is easy to monitor and fix in bulk with SQL, for example:

```
select * from wp_posts where comment_status != 'open' and post_type IN ('product', 'post');

update wp_posts set comment_status = 'open' where post_type IN ('product', 'post') AND comment_status != 'open' ;

select * from wp_posts where comment_status != 'open' and post_type IN ('product', 'post');
```

As soon as comment_status is set back to open comments become enabled again.

**Comment by Jon007** [comment]
Polylang synchronisation of comment_status field works fine for posts.

This sounds like an event sequence issue, maybe $post->comment_status is no longer set when the post is the product, as product only loads into $product->reviews_allowed, so then when Polylang copies the post properties it is not set..

**Comment by Jon007** [comment]
Root cause is noted on woocommerce issue: https://github.com/woocommerce/woocommerce/issues/16153

A workaround could be to hook the save before the pll_save_post is called and set:
$post->comment_status = ($product->get_reviews_allowed()) ? 'open' : 'closed';

In the meantime I suggest to turn off the Polylang synchronisation setting at:
 Languages, Settings, Synchronisation, Comment Status

**Comment by Jon007** [comment]
this seems to be fixed on the woocommerce side but a retest says it isn't fixed in 3.1.2 so needs retesting with 3.2.5

**Comment by hyyan** [comment]
@Jon007 This one seems fixed and works fine for me with 

**Polylang** : v2.3.4
**Woocommerce**: v3.3.5
**WooPoly** : master

Please feel free to re-open if stills an issue

**Comment by Jon007** [comment]
oh yes, this is probably fixed by the same Polylang change as #298
